src/components/TradeWidget/index.tsx

```diff
@@ -313,10 +314,17 @@ const TradeWidget: React.FC<TradeWidgetProps> = ({
     )
   }, [dispatch, priceValue, sellValue, sellToken, receiveToken, validFromValue, validUntilValue])
 
+  // Get canonical market
+  const { baseToken, quoteToken } = useMemo(() => getMarket({ receiveToken, sellToken }), [receiveToken, sellToken])
+
   // Update receive amount
   useEffect(() => {
-    priceValue && sellValue && setValue(receiveInputId, calculateReceiveAmount(priceValue, sellValue))
-  }, [priceValue, priceInverseValue, setValue, sellValue])
+    if (priceValue && sellValue) {
+      // If price is quoted in sell tokens, we use it, otherwise we use the inverse
+      const priceUsedForReciveAmount = sellToken === quoteToken ? priceValue : priceInverseValue
```

Comment:
why not just use `wasPriorityAdjusted` here? isn't that the same thing

Reply:
Cause this is easier to reason about. In the old PR `wasPriorityAdjusted ` was being drilled down to components, and is not super clear what it means. 

wasPriorityAdjusted + isPriceInverse + sell/buy tokens, quote/base tokens ---> 🤯

Here it's easier to read:
* if sell token is the quote, I use the price
* otherwise, I use the inverse 

I don't care if user flipped the market, or the smart logic did it, I care about the units of the price